examples/ui/grid.rs

```diff
@@ -0,0 +1,218 @@
+                        align_items: AlignItems::Start,
+                        // Align content towards the center in the horizontal axis
+                        justify_items: JustifyItems::Center,
+                        // Add 20px padding to the top
+                        padding: UiRect::top(Val::Px(20.)),
+                        ..default()
+                    },
+                    background_color: BackgroundColor(Color::BLACK),
+                    ..default()
+                })
+                .with_children(|builder| {
+                    builder.spawn(TextBundle::from_section(
+                        "Sidebar",
+                        TextStyle {
+                            font,
+                            font_size: 24.0,
+                            color: Color::WHITE,
+                        },
+                    ));
+                });
+
+            // Footer / status bar
+            builder.spawn(NodeBundle {
+                style: Style {
+                    // Make this node span two grid column so that it takes up the entire bottom row
+                    grid_column: GridPlacement::span(2),
+                    ..default()
+                },
+                background_color: BackgroundColor(Color::WHITE),
+                ..default()
+            });
+
+            // Modal (absolutely positioned on top of content - uncomment to view)
+            // builder.spawn(NodeBundle {
+            //     style: Style {
+            //         position_type: PositionType::Absolute,
+            //         margin: UiRect {
+            //             top: Val::Px(100.),
+            //             bottom: Val::Auto,
+            //             left: Val::Auto,
+            //             right: Val::Auto,
+            //         },
+            //         size: Size {
+            //             width: Val::Percent(60.),
+            //             height: Val::Px(300.),
+            //         },
+            //         max_size: Size {
+            //             width: Val::Px(600.),
+            //             height: Val::Auto,
+            //         },
+            //         ..default()
+            //     },
+            //     background_color: BackgroundColor(Color::Rgba {
+            //         red: 255.0,
+            //         green: 255.0,
+            //         blue: 255.0,
+            //         alpha: 0.8,
+            //     }),
+            //     ..default()
+            // });
+        });
+}
+
+/// Create a coloured rectangle node. The node has size as it is assumed that it will be
+/// spawned as a child of a Grid container with `AlignItems::Stretch` and `JustifyItems::Stretch`
+/// which will allow it to take it's size from the size of the grid area it occupies.
+fn item_rect(builder: &mut ChildBuilder, color: Color) {
+    builder
+        .spawn(NodeBundle {
+            style: Style {
+                display: Display::Grid,
```

Comment:
what does using display: Grid without declaring any rows or columns do?

Reply:
It creates a grid with a single row and a single column, both of which have `Auto` size (if you add more than child to it then additional `Auto` sized rows with be implicitly created to fit them). I think in this case it is being used to stretch it's child element to fit in both axis. With a flexbox container, you'd need to set `flex-grow: 1` on the child to achieve this effect as it only stretches in the cross axis by default.